core/includes/form.inc

```diff
@@ -4883,13 +4942,13 @@ function theme_form_element($variables) {
   // Add element's #type and #name as class to aid with JS/CSS selectors.
   $attributes['class'][] = 'form-item';
   if (!empty($element['#type'])) {
-    $attributes['class'][] = 'form-type-' . strtr($element['#type'], '_', '-');
+    $attributes['class'][] = backdrop_html_class('form-type-' . $element['#type']);
```

Comment:
Do we know for sure that this change will result in equivalent classes? This seems like a potentially breaking change.

Reply:
Lets see:
- the implementation of `strtr()` simply replaces underscores with dashes.
- `backdrop_html_class()` does the following:
  - passes everything through `backdrop_strtolower()`, and according to https://stackoverflow.com/questions/12533926/are-class-names-in-css-selectors-case-sensitive:
    - CSS selectors are generally case-insensitive; this includes class and ID selectors 👍🏼 
    - But [HTML class names are case-sensitive](http://www.w3.org/TR/html4/struct/global.html#h-7.5.2) (see the attribute definition) ... This has not changed in [HTML5](http://www.w3.org/TR/html50/disabled-elements.html#case-sensitivity).1 👎🏼 
  - passes everything through `backdrop_clean_css_identifier()`, which converts underscores (among other things) to dashes.

So yes, this is a potentially breaking change, **but only in theory**. For that to happen, someone needs to have declared `'#type'` for an element using uppercase letters, in which case, I believe that there would be errors thrown or the type would simply not be recognized. Since the scope of the `$element['#type']` that is passed to `strtr()` is limited (only element types that have been officially specified in the Form API), I believe that this is safe to do, and it provides standardization and future-proofing.